libcontainer/init_linux.go

```diff
@@ -196,10 +195,6 @@ func startInitialization() (retErr error) {
 		dmzExe = os.NewFile(uintptr(dmzFd), "runc-dmz")
 	}
 
-	// clear the current process's environment to clean any libcontainer
-	// specific env vars.
-	os.Clearenv()
```

Comment:
How about remaining this? Is this affect the performance?

Reply:
Unfortunately it does affect performance. What it does internally is calls Unsetenv for every variable (and Unsetenv calls glibc's `unsetenv(3)`). You can add this back and use the benchmark provided to see what difference does it make (I haven't checked it but I checked Clearenv in other benchmarks and it is slow).

Also, while I understand the security concert, with this PR clearenv is definitely not needed because now the container process does not use any environment from the parent process.

Reply:
Damn you're right, Go has their own copy of the environment and they implement it that way (`clearenv(3)` is very efficient by comparison). It seems the reason they clear the environment that way is because they need to detect changes to `GODEBUG` at runtime -- but this could be done more efficiently (by clearing the environment properly and then just checking whether the old environment had a `GODEBUG` value and then calling the unset handler for just that value).

I don't think there is a practical issue with dropping it, but I would prefer to fix the Go stdlib issue rather than dropping this. I don't want some other change to result in us calling out to a binary and we accidentally leak environment variables from the host.

<hr>

Funnily enough, this code was originally added in 44636dd8632756a70400793d84aeab3673a8d057 AFAICS, where the way they did re-exec at the time meant they moved from using `command.Env` to `Clearenv`+`Setenv`. :smile_cat:

Reply:
Yes, I also prefer to keep this, because there are usually only no more than 10 envs here.